crates/gen-host-js/src/lib.rs

```diff
@@ -729,8 +784,22 @@ impl Instantiator<'_> {
         // so the instantiation should be relatively straightforward.
         let i = self.instances.push(idx);
         let name = format!("module{}.wasm", idx.as_u32());
-        uwrite!(self.src.js, "const instance{} = ", i.as_u32());
-        uwriteln!(self.src.js, "await instantiateCore(\"{name}\", {imports});");
+        let iu32 = i.as_u32();
+        if self.gen.opts.instantiation {
+            uwriteln!(
+                self.src.js,
+                "const instance{iu32} = await instantiateCore(\"{name}\", {imports});"
+            );
+        } else {
+            let load_wasm = self.gen.intrinsic(Intrinsic::LoadWasm);
+            uwriteln!(
+                self.src.js,
+                "
+                const instance{iu32} = await {load_wasm}(new URL('./{name}', import.meta.url))
+                .then(m => WebAssembly.instantiate(m, {imports}));\
+            "
```

Comment:
I think this string literal may need one tab of indentation to match.

Additionally, is it worth trying to use `instantiateStreaming` instead of just `compileStreaming` since I think at least historically that's what browsers recommended? The `LoadWasm` intrinsic here could perhaps be replaced with `InstantiateWasm` where the imports are passed in.

Reply:
Added, although as far as I can tell from the source formatter, it will remove that indentation anyway?

In theory if we wanted to instantiate the same core module multiple times, we would want the compiled but uninstantiated module. For this reason I quite like treating that as the go-to representation, and since it matches what will happen with import reflection as well.

It was just the sync former APIs which are not - that is `WebAssembly.instantiate` is not recommended when passing a buffer. But that it's overloaded form that takes a Module instance is fine to use despite the MDN warning, since the steps involved are equivalent from a spec perspective at least.